cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,171 @@ def test_image_url_is_shown_on_movie_session_detail(self):
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class MovieUnauthorizedTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_is_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class MovieAuthorizedTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="user@user.com",
+            password="password",
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movies(self):
+        """Test retrieving a list of movies"""
+        sample_movie()
+        movie = sample_movie()
+
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        res = self.client.get(MOVIE_URL)
+
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_list_movies_with_filter(self):
+        """Test retrieving a list of movies with filter"""
+        sample_movie()
+        movie = sample_movie()
+
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        res = self.client.get(MOVIE_URL, {"title": "Title"})
+
+        movies = Movie.objects.filter(title="Title")
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_list_movies_with_filters(self):
+        """Test retrieving a list of movies with filters"""
+        sample_movie()
+        movie = sample_movie()
+
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        res = self.client.get(MOVIE_URL, {"title": "Title", "genres": 1})
+
+        movies = Movie.objects.filter(title="Title", genres=1)
+        serializer = MovieListSerializer(movies, many=True)
```

Comment:
again.  Sample movies, than add needed genres and use your `self.client.get()` to filter movies by needed genre. Then serialize those previously created movies and check if data in response is equal to serialized data